### PR TITLE
Added the ProvisionCommand class

### PR DESCRIPTION
**Pull request by andre-sa**
I've created the ProvisionCommand class so the Homestead machine could be provisioned without the need for shutdown or restart.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /vendor
 /.vagrant
+/.idea
```
Comment (GrahamCampbell) [inline .gitignore]: please remove that

Comment (GrahamCampbell) [inline .gitignore]: ide ignoring should be done at the system level

**Comment by andrewhood125** [comment]
Hey @andre-sa, you might want to do a `git reset --soft` back to 
when you added the ProvisionCommand. Then you can remove `.idea` from the `.gitignore` and redo your original commit so the contents of your `.idea` directory are not included in Homesteads history. Don't forget to system wide ignore `.idea`!

**Comment by andre-sa** [comment]
I believe i've done it now. Thank for your help @GrahamCampbell and @andrewhood125.

**Comment by milesj** [comment]
You misspelled provision, `provisiom`.

**Pull request closed without merging**